src/mmw/js/src/data_catalog/models.js

```diff
@@ -81,38 +165,59 @@ var Catalog = Backbone.Model.extend({
         }
     },
 
-    search: function(query, fromDate, toDate, geom) {
+    search: function(query, geom) {
         this.set({
             query: query,
             geom: geom,
-            fromDate: fromDate,
-            toDate: toDate,
         });
 
         return this.startSearch(1);
     },
 
+    isSearchValid: function() {
+        var query = this.get('query'),
+            dateFilter = this.get('filters').findWhere({ id: 'date'}),
```

Comment:
This can be made more generic, since each filter can have a `validate()` method that returns `true` or `false`:

```js
var query = this.get('query'),
    validate = function(x) { return x.validate(); },
    valid = this.get('filters').map(validate);

return query && _.every(valid);
```

For simpler filters like `GriddedService` which is just a checkbox, we can have its `validate()` always return true.